### ISSUE TITLE
Facebook Example

### ISSUE THREAD

**Issue by josephkandi** [MicrosoftDocs/xamarin-docs#68]
It would be nice if you can include the Facebook example as well, since they are slightly different in how you would configure the Android portion of the Intent filters and also considering that Facebook automatically provides the token without making an additional API request 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: aa8bb9e0-5898-0d11-0213-fd4d8c8d0297
* Version Independent ID: 2f46e1f7-164f-2f32-70c0-f28919d16c5c
* Content: [Authenticating Users with an Identity Provider - Xamarin](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/data-cloud/authentication/oauth)
* Content Source: [docs/xamarin-forms/data-cloud/authentication/oauth.md](https://github.com/MicrosoftDocs/xamarin-docs/blob/live/docs/xamarin-forms/data-cloud/authentication/oauth.md)
* Service: **unspecified**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

**Comment by conceptdev** [comment]
There are additional authentication examples in the Azure docs, including https://docs.microsoft.com/en-us/azure/app-service-mobile/app-service-mobile-xamarin-android-get-started-users

Thanks for your feedback.

**Comment by stevehurcombe** [comment]
You might be interested to know that Facebook no longer support custom URI schemes in their callback URLs. This will have implications for Xamarin.Auth and the documentation - as I had been following it and failed at the final hurdle!

"_Thanks for getting in touch. We'll make the messaging more clear on this, but we do not support custom URI schemes as valid redirect URIs. To perform login in a native app without opening the browser, please use the official SDKs (is https://github.com/xamarin/FacebookComponents what you're looking for?)._"
https://developers.facebook.com/bugs/351288075391305/

**Comment by conceptdev** [comment]
thanks for the additional info @stevehurcombe

**Comment by moljac** [comment]
Dear Steve

Your statement depends on context - you cannot define your OWN custom url scheme, but if you check this:

https://github.com/facebook/facebook-android-sdk/blob/45482361eb182bb3392386182f8f4e6c4896c2b9/facebook-common/src/main/java/com/facebook/CustomTabMainActivity.java#L45

and

https://github.com/facebook/facebook-ios-sdk/blob/master/FBSDKLoginKit/FBSDKLoginKitTests/FBSDKLoginManagerTests.m#L51

Analysing this code `redirect_url` is in the form: 

```csharp
string redirect_url = $"fb{fb_app_id}://authorize";
```
where fb_app_id is your AppId from FB portal.

`fb3289302138293821` is custom scheme and still works.

**Comment by josephkandi** [comment]
Thanks @moljac. For Facebook that's exactly how the scheme is supposed to be, and it seems thats not documented anywhere other than digging through the code.

**Comment by stevehurcombe** [comment]
Not any more it would seem. I read the same, I also read detailed.readme in Xamarin.Auth (I think). 

That didn't work though, so I sent a support request to Facebook with exactly that format and that's the response I got. 

They've tightened the process up - now you have to whitelist your callback URLs in their settings and that format generates an error in their UI.

**Comment by moljac** [comment]
Steve,
This is another problem, server side (backend) for most OAuth providers is never ending story... And it is frustrating to keep up and document, because changes happen quite often.

> hey've tightened the process up - now you have to whitelist your callback URLs in their settings and that format generates an error in their UI.

True, you cannot whitelist `fb<app_id>://authorize`, because it generates error, but it is their default and I did not have problems with it. I admit I did not test FB in last 2 weeks. 
In OAuth urls `fb<app_id>` is sent for sure (otherwise their SDK code would not work too). All I can think of they might send special UserAgent from SDK code and check that on server side. If that is true, this might be a problem, because there is no way to influence UserAgent from CustomTabs and/or SFSafariViewController.
